spec/coffee/directives/api/circle.spec.coffee

```diff
@@ -0,0 +1,129 @@
+describe 'uiGmapCircle', ->
+  allDone =  undefined
+  rootScope = null
+  timeout = null
+  GCircle = null
+  digest = null
+  modelClicked = false
+  browser = null
+
+  afterEach ->
+    digest = null
+    GCircle.resetInstances()
+    modelClicked = false
+
+  beforeEach ->
+    digest = (fn) =>
+     fn()
+     timeout?.flush() if browser.deferredFns?.length
+     rootScope?.$digest()
+     
+    @html = """
+      <ui-gmap-google-map draggable="true" center="map.center" zoom="map.zoom">
+          <ui-gmap-circle center='map.circle.center'
+                        radius='map.circle.radius'
+                        fill='map.circle.fill'
+                        stroke='map.circle.stroke'
+                        clickable='map.circle.clickable'
+                        draggable='map.circle.draggable'
+                        editable='map.circle.editable'
+                        visible='map.circle.visible'
+                        events='map.circle.events'>
+        </ui-gmap-circle>
+      </ui-gmap-google-map>
+    """
+    @map =
+      zoom: 12
+      center : {longitude: 47, latitude: -27}
+
+    @circle =
+      id: 1,
+      center: {longitude: 47, latitude: -27},
+      radius: 500,
+      stroke:
+        color: '#08B21F',
+        weight: 2,
+        opacity: 1
+      fill:
+        color: '#08B21F',
+        opacity: 0.5
+      geodesic: true
+      draggable: true
+      clickable: true
+      editable: true
+      visible: true
+      events:
+        radius_changed: (gObject) ->
+
+
+    apiMock = window['uiGmapInitiator'].initMock().apiMock
+    GCircle = window.google.maps.Circle
+
+    inject ['$rootScope', '$timeout', '$compile', '$q', 'uiGmapCircle', '$browser',
```

Comment:
I really hate angulars inject... it is not DRY at all. We should be able to use it more than once, where I can do the rootScope timeout.. etc crap once. And then the stuff I need like uiGmapCircle later.